lib/coinbase/staking_operation.rb

```diff
@@ -4,18 +4,37 @@ module Coinbase
   # A representation of a staking operation (stake, unstake, claim rewards, etc). It
   # may have multiple steps with some being transactions to sign, and others to wait.
   # @attr_reader [Array<Coinbase::Transaction>] transactions The list of current
+  # @attr_reader [Symbol] status The status of the operation
+  # @attr_reader [String] error The error message if the operation failed
   #   transactions associated with the operation
   class StakingOperation
-    attr_reader :transactions
+    attr_reader :transactions, :status, :error
+
+    # Returns all StakingOperations.
+    # @return [Enumerable<Coinbase::StakingOperation>] The staking operations
+    def self.list
+      Coinbase::Pagination.enumerate(
+        ->(page) { Coinbase.call_api { Coinbase::Client::StakingOperation.list(page: page) } }
+      ) do |staking_operation|
+        new(staking_operation)
+      end
+    end
+
+    # Returns the StakingOperation with the provided ID.
+    # @param id [String] The ID of the StakingOperation
+    # @return [Coinbase::StakingOperation] The staking operation
+    def self.fetch(id)
+      new(load_from_server(id))
+    end
+
+    def self.load_from_server(id)
```

Comment:
Can we just inline this since it's only ever used in `self`? Or make this a `private_class_method`?

Reply:
got rid of load_from_server and started doing stuff inline.